### DISCUSSION TITLE
Multiple codecs requires multiple connections?

### DISCUSSION THREAD

**Discussion by apanloco** [redis/lettuce#2108] [Q&A]
Hey!

I am reading/writing data in Redis using four codecs:
- <String, byte[]> with gzip
- <String, byte[]> plain, no gzip
- <String, String> with gzip
- <String, String> plain, no gzip

(Just to be super clear, with gzip I mean something like this: https://github.com/lettuce-io/lettuce-core/blob/main/src/main/java/io/lettuce/core/codec/CompressionCodec.java)

As I understand the API i need four different connections to do be able to use four different codecs. Is this correct? Or is there some way to reuse the same connection with multiple codecs? Thanks!

PS: Latest Lettuce.

**Comment by mp911de** [comment]
Yep, that's currently a limitation. Alternatively, you can use different codecs when issuing custom commands via the `dispatch(…)` method with some casting trickery if you do not use Pub/Sub on the same connection.
